figures/tasks.py

```diff
@@ -328,6 +374,63 @@ def populate_daily_metrics_next(site_id=None, force_update=False):
                            site_count=sites_count))
 
 
+@shared_task
+def daily_metrics_callback(results, *args):
+    logger.info(results)
+    logger.info(args)
+
+
+@shared_task
+def populate_daily_metrics_async_sites(site_id=None, force_update=False):
+    """Next iteration to collect daily metrics for all sites in a deployment
+
+    This is a top level Celery task run every 24 hours to update Figures data.
+
+    * It updates Figures per-enrollment data and collect daily aggregate metrics
+    * It's purpose is to collect new metrics on an ongoing basis and not serve
+      dual purpose of collecting ongoing data AND backfilling data.
+    * The driver for this change is to improve performance of the daily Celery jobs
+
+    What's different?
+
+    * Figures collects the enrollment data first, then aggregates daily data.
+
+    TODO: Draft up public architecture docs and reference them here
+    """
+    if waffle.switch_is_active(WAFFLE_DISABLE_PIPELINE):
+        logger.warning('Figures pipeline is disabled due to %s being active.',
+                       WAFFLE_DISABLE_PIPELINE)
+        return
+
+    date_for = datetime.datetime.utcnow().date()
```

Comment:
I guess I'll leave my usual comment that this function would be easier to test and more generally useful if `date_for` is passed as a parameter with default `None` and only set to the current date when unset, so:

```
def populate_daily_metrics_async_sites(site_id=None, force_update=False, date_for=None):
...
if not date_for:
    date_for = datetime.datetime.utcnow().date()
```